services/src/main/java/io/druid/cli/EmbeddedBroker.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.cli;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+import com.metamx.common.guava.Sequence;
+import com.metamx.common.lifecycle.Lifecycle;
+import io.druid.guice.GuiceInjectors;
+import io.druid.query.Query;
+import io.druid.query.QueryContextKeys;
+import io.druid.query.QuerySegmentWalker;
+import io.druid.server.QueryManager;
+import io.druid.server.initialization.ServerConfig;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ */
+public class EmbeddedBroker implements Closeable
+{
+  public static EmbeddedBroker create() throws Exception
+  {
+    final Injector injector = GuiceInjectors.makeStartupInjector();
+    CliBroker broker = new CliBroker();
+    injector.injectMembers(broker);
+
+    Injector brokerInjector = broker.makeInjector();
```

Comment:
Do we need all modules from CliBroker? Seems to me that this class is mainly for running and canceling queries by directly calling `runQuery` and `cancelQuery`. In that case, I think all Jetty-related stuff (e.g. `QueryJettyServerInitializer`, `QueryResource`) can be removed.